### ISSUE TITLE
Tree Widget

### ISSUE THREAD

**Issue by Yoda12999** [PistonDevelopers/conrod#869]
It would be lovely to have a tree view (see QTreeWidget in Qt or the tree in dear imgui) for file views or hierarchical views in general.

**Comment by mitchmindtree** [comment]
@Yoda12999 Hey there, thanks for the issue!

We have a [FileNavigator](http://docs.piston.rs/conrod/conrod/widget/file_navigator/struct.FileNavigator.html) (see #737 for pics or run `examples/file_navigator.rs`) which sounds like it might be similar to what you're after? I like your idea of abstracting it into a more general `Tree` widget though, as there are almost certainly more use cases for it then navigating files :+1:

**Comment by Yoda12999** [comment]
Yes, FileNavigator is wonderful for traversing a file hierarchy to select a file, but even for just displaying a total directory structure, such as a project overview, it falls flat.
